.github/workflows/mutexbot.yml

```diff
@@ -0,0 +1,122 @@
+name: MutexBot Container Image
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "mutexbot/**"
+  push:
+    branches:
+      - main
+    paths:
+      - "mutexbot/**"
+    tags:
+      - "mutexbot-v*.*.*"
+
+permissions:
+  contents: read
+
+env:
+  DOCKERHUB_REPO: neondatabase/dev-actions
+
+jobs:
```

Comment:
I see these jobs have some additional steps/differences in comparison to what we usually do, for example, in https://github.com/neondatabase/neon/blob/main/.github/workflows/build-build-tools-image.yml#L87-L161 or in https://github.com/neondatabase/neon/blob/b9464865b619d35257500bfade7651e3adfc07e4/.github/workflows/build_and_test.yml#L542-L626

I prefer the same approach to make it easier to understand/support.
(if we want to change the approach in all repos, that probably should be a separate issue first)

Reply:
I've done this based on the docs in https://docs.docker.com/build/ci/github-actions/multi-platform/, after looking at https://github.com/neondatabase/gh-workflow-stats-action/blob/main/.github/workflows/ci.yml first for inspiration. I'll compare this to what we do in other places and will try making it closer to those.

I think in the long run it will make sense to have a reusable workflow that handles registry login and image building and so on where you only have to set a few vars, so that we don't define complex workflows over and over again.

Reply:
https://github.com/neondatabase/gh-workflow-stats-action/blob/main/.github/workflows/ci.yml should be unified as well at some point :)

> I think in the long run it will make sense to have a reusable workflow that handles registry login and image building and so on where you only have to set a few vars, so that we don't define complex workflows over and over again.

agree

Reply:
After thinking about it more: I think it does make sense to not clutter the registry with unnecessary tags, so having separate images with architecture suffixes is something we should stop at some point. I will keep this difference to the other places for now. Switching from artifacts to outputs for digest passing reduced the amount of steps we have, and we got closer to how images are built in other places.

I think there's room for improvement still, to make things more consistent, but that's mostly a thing of creating that reusable workflow and making sure we use that everywhere. I'd like to postpone that and merge this as is.